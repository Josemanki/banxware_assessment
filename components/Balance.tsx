import React from 'react';
import { TBalance } from '../types';

type Props = {
  balance: TBalance;
};

const Balance = ({ balance }: Props) => {
  return (
    <div className="balance">
      <span className="balance__amount">{balance.amount.toLocaleString()}</span>{' '}
      <span className="balance__currency">{balance.currency}</span>
    </div>
  );
};

export default Balance;
