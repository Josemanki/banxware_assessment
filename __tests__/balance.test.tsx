import { render } from '@testing-library/react';
import Balance from '../components/Balance';
import '@testing-library/jest-dom';

jest.mock('../components/Chart');

const mockBalance = { amount: 12000, currency: 'EUR' };

test('Should render the balance component', () => {
  const { asFragment } = render(<Balance balance={mockBalance} />);
  expect(asFragment()).toMatchSnapshot();
});

test('Should render balance with the right amount', () => {
  const { getByText } = render(<Balance balance={mockBalance} />);
  expect(getByText('12.000')).not.toBeNull();
});

test('Should render currency with the right one', () => {
  const { getByText } = render(<Balance balance={mockBalance} />);
  expect(getByText('EUR')).not.toBeNull();
});
