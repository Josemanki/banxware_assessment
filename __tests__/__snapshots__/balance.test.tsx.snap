// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the balance component 1`] = `
<DocumentFragment>
  <div
    class="balance"
  >
    <span
      class="balance__amount"
    >
      12.000
    </span>
     
    <span
      class="balance__currency"
    >
      EUR
    </span>
  </div>
</DocumentFragment>
`;
